### PR TITLE
feat: custom log directory

### LINKED ISSUE #464: Spark History Server support for Azure Data Lake Storage Gen2 (ADLS)
Spark application logs should be storable in Azure Data Lake Storage Gen2 (ADLS).

The [`hadoop-azure`](https://hadoop.apache.org/docs/stable/hadoop-azure/) module must be added to the Spark image and the `spec.logFileDirectory` structure must be extended with `adls`. Currently only `s3` is supported. Alternatively, a workaround by setting `sparkConf` accordingly could be documented.

The SAS token should be read from a Secret.

- [x] #473 
- [x] stackabletech/decisions#32
- [ ] #479 (including an integration test for a custom log directory in HDFS)
- [x] Test manually a custom log directory in ADLS with `spark-k8s-operator:0.0.0-pr479`
  SparkHistoryServer: https://github.com/stackabletech/sbernauer-customers/blob/e2e6db97645ac2d0fd54130c81e45a8d3e7107e7/airplus/poc-cluster/specs/spark-history.yaml
  SparkApplication: https://github.com/stackabletech/sbernauer-customers/blob/e2e6db97645ac2d0fd54130c81e45a8d3e7107e7/airplus/poc-cluster/test/spark.yaml

### PR DESCRIPTION
**Pull request by siegfriedweber**
# Description

Allow specifying a custom log directory

Closes #464 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved (stackabletech/decisions#32)
- [x] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] Code contains useful logging statements
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: docs/modules/spark-k8s/pages/index.adoc [hunk 1]
```diff
@@ -37,8 +37,8 @@ The SparkApplication resource is the main point of interaction with the operator
 An exhaustive list of options is given in the {crd}[SparkApplication CRD reference {external-link-icon}^].
 
 The xref:usage-guide/history-server.adoc[SparkHistoryServer] has a single `node` role.
-It is used to deploy a https://spark.apache.org/docs/latest/monitoring.html#viewing-after-the-fact[Spark history server] that displays application logs from S3 buckets.
-Of course, your applications need to write their logs to the same buckets.
+It is used to deploy a https://spark.apache.org/docs/latest/monitoring.html#viewing-after-the-fact[Spark history server] that displays application logs.
+Of course, your applications need to write their logs to the location.
```
Comment (razvan) [inline docs/modules/spark-k8s/pages/index.adoc:41]:
```suggestion
Of course, your applications need to write their logs to the same location.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ All notable changes to this project will be documented in this file.
 ### Added
 
 - Make spark-env.sh configurable via `configOverrides` ([#473]).
+- Allow specifying a custom log directory ([#479]).
```
Comment (razvan) [inline CHANGELOG.md:10]: I think we need to be more clear:

```suggestion
- The Spark history server can now service logs from HDFS compatible systems ([#479]).
```

**Review by razvan — APPROVED** [review]
LGTM with some suggestions.

File: CHANGELOG.md (hunk 2 above)
Comment (siegfriedweber) [reply CHANGELOG.md:10]: You can enter an arbitrary file system in `customLogDirectory` as long as the required JARs are there. For instance, you could configure S3 (which is not HDFS compatible) with `customLogDirectory`.

Comment (razvan) [reply CHANGELOG.md:10]: I know but "custom log directory" doesn't say anything about different storage systems.

You don't have to accept my suggestion but  I would like for it to be more specific.

**Review by razvan — APPROVED** [review]

**Pull request merged**